mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -351,7 +351,14 @@ object LDA {
     val nweGraph = graph.mapTriplets(
       (pid, iter) => {
         val gen = new XORShiftRandom(parts * innerIter + pid)
-        val wordTableCache = new AppendOnlyMap[VertexId, SoftReference[(Double, Table)]]()
+        // table is a per term data structure
+        // in GraphX, edges in a partition are clustered by source IDs (term id in this case)
+        // so, use below simple cache to avoid calculating table each time
+        val lastTable = (new ArrayBuffer[Int](numTopics.toInt),
+                         new ArrayBuffer[Int](numTopics.toInt),
```

Comment:
This will not improve performance?  The array field in [ArrayBuffer.scala](https://github.com/scala/scala/blob/2.10.x/src/library/scala/collection/mutable/ArrayBuffer.scala) is `Array[AnyRef]`